retry/options.go

```diff
@@ -29,6 +32,13 @@ func WithJitter(o time.Duration) Option {
 	}
 }
 
+// WithErrorLog logs errors as they are encountered during the retry loop.
+func WithErrorLog(enable bool) Option {
```

Comment:
nit:  WithErrorLogging rather than WithErrLog, since you are not supplying the log with this function.